packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1736,12 +1736,12 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      * using IDs generated locally. After attach, these IDs cannot be used, so this table maps the old local IDs to the
      * new storage IDs so requests can be redirected.
      */
-    public createSummary(blobRedirectTable?: Map<string, string>): ISummaryTree {
+    public async createSummary(blobRedirectTable?: Map<string, string>): Promise<ISummaryTree> {
         if (blobRedirectTable) {
             this.blobManager.setRedirectTable(blobRedirectTable);
         }
 
-        const summarizeResult = this.dataStores.createSummary();
+        const summarizeResult = await this.dataStores.createSummary();
```

Comment:
Similar problem here (requiring changing existing flows / APIs) - addContainerBlobsToSummary() below should happen before await here, as we need to do all "initiation" of data collection synchronously in one go, to make sure it represents state at the same point in time.